src/daemon/daemon.cpp

```diff
@@ -1295,6 +1306,24 @@ void populate_snapshot_info(mp::VirtualMachine& vm,
     populate_snapshot_fundamentals(snapshot, fundamentals);
 }
 
+std::string generate_netplan_script(int index, const std::string& mac_address)
+{
+    return fmt::format("echo -e \""
+                       "# Generated by Multipass\n"
+                       "network:\n"
+                       "    ethernets:\n"
+                       "        extra{0}:\n"
+                       "            dhcp4: true\n"
+                       "            dhcp4-overrides:\n"
+                       "                route-metric: 200\n"
+                       "            match:\n"
+                       "                macaddress: {1}\n"
+                       "            optional: true\n"
+                       "    version: 2"
+                       "\" | sudo dd of=/etc/netplan/51-extra{0}.yaml oflag=append conv=notrunc",
```

Comment:
Could we perhaps begin the filename with "multipass" (i.e. `multipass-extra{0}.yaml`), so that it is immediately obvious we created it?

Reply:
Good idea. However, I kept the `51-` to force it to run after the cloud-init file, which is `50-cloud-init.yaml`.